tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -781,14 +868,16 @@ Status TRTEngineOp::GetEngineCacheResource(OpKernelContext* ctx,
 }
 
 StatusOr<EngineContext*> TRTEngineOp::GetEngine(
-    const std::vector<TensorShape>& input_shapes, OpKernelContext* ctx,
+    const std::vector<TensorShape>& input_concrete_shapes, OpKernelContext* ctx,
     TRTEngineCacheResource* cache_res) {
   static EngineContext empty_context;
 
   mutex_lock lock(engine_mutex_);
   // Using first input to get batch size is reliable - VerifyInputShapes() has
-  // verified that.
-  const int batch_size = input_shapes[0].dim_size(0);
+  // verified that. Although this is not needed for explicit batch mode, but
+  // we still have to pass it to ConvertGraphDefToEngine until the requirement
+  // is removed from that function.
+  const int batch_size = input_concrete_shapes[0].dim_size(0);
```

Comment:
Can you change the comment to this:
VerifyInputShapes guarantee that the first input is not a scalar. As such we can always use the first input to get the batch size for implicit batch mode. For explicit batch mode, this value is not used. 
Then add a TODO for removing the need to batch size to ConvertGraphDefToEngine for explicit batch mode.